ls8/cpu.c

```diff
@@ -59,15 +59,45 @@ void alu(struct cpu *cpu, enum alu_op op, unsigned char regA, unsigned char regB
 void cpu_run(struct cpu *cpu)
 {
   int running = 1; // True until we get a HLT instruction
+  int PC = cpu->PC;
 
   while (running) {
     // TODO
     // 1. Get the value of the current instruction (in address PC).
+    unsigned char instruction = data[PC];
     // 2. Figure out how many operands this next instruction requires
+    
     // 3. Get the appropriate value(s) of the operands following this instruction
+    switch(instruction){ 
+      case LDI:
+        unsigned char ldi_reg = data[cpu->PC+1];
```

Comment:
you can merge these into the same `switch()` case, so you're not calling `switch()` twice.